gapis/trace/tracer/tracer.go

```diff
@@ -49,6 +51,18 @@ type Process interface {
 	Capture(ctx context.Context, start task.Signal, stop task.Signal, w io.Writer, written *int64) (size int64, err error)
 }
 
+type GpuCounter struct {
+	Id    uint32
+	Name  string
+	Check func(column *perfetto_service.QueryResult_ColumnValues, columnType perfetto_service.QueryResult_ColumnDesc_Type) bool
+}
+
+type Validator interface {
```

Comment:
I think it's better to move this down a level. That is, from the `Tracer`'s perspective, there is just a simple `Validate()` function. This moves all the validation logic to a lower layer. I think validation of a desktop or Stadia device will look drastically different from an Android device.

For now, desktop and Stadia can just have the `Validate()` function return a successful result, while the Android validator can do the whole tracing-processing-query validation loop that is aware of counters, etc.